viewing_party/party.py

```diff
@@ -0,0 +1,184 @@
+# Wave 1
+
+def create_movie(movie_title, genre, rating):  
+  new_movie = {}
+  
+  if not movie_title or not genre or not rating:
+    return None  
+  
+  else:
+    new_movie["title"] = movie_title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+    
+  return new_movie
+
+
+def add_to_watched(user_data, movie):
+
+  if not "watched" in user_data:
+    user_data["watched"] = [movie]
+    
+  else: 
+    user_data["watched"].append(movie)
+    
+  return user_data
+  
+  
+def add_to_watchlist(user_data, movie):
+  updated_dict = user_data
+  
+  if not "watchlist" in user_data:
+    user_data["watchlist"] = [movie]
+    
+  else: 
+    user_data["watchlist"].append(movie)
+    
+  return user_data
+
+
+def watch_movie(user_data, title):
+  
+  movie_found = False
+  index_to_be_removed = -1  # set to -1 bcs invalid, can't set it to other bcs those would be valid
+  
+  for index, temp_movie_dict in enumerate(user_data["watchlist"]):
+    if temp_movie_dict["title"] == title: #found movie to be removed
+      index_to_be_removed = index #the index of the movie
+      movie_found = True
+      user_data["watched"].append(temp_movie_dict)
+    
+  if movie_found:
+    user_data["watchlist"].pop(index_to_be_removed)
+  
+  return user_data
+
+
+# Wave 2
+
+def get_watched_avg_rating(user_data):
+  user_rating_total = 0.0
+  average = 0.0
+  watched_list = user_data["watched"]
+  
+  if watched_list != []:
+    for movie_dict in watched_list:
+      user_rating_total += movie_dict["rating"] 
+    average = user_rating_total / len(watched_list)
+    
+  return average
+
+      
+def get_most_watched_genre(user_data):
+  popular_genre_list = []
+  count = 0
+  watched_list = user_data["watched"]
+  popular_genre = None
+  
+  if watched_list != []:
```

Comment:
This if check seems unnecessary. If the list is empty, then the following loop will run 0 times, so `popular_genre_list` would stay empty, so then the genre counting would also run 0 times, and the `popular_genre` would keep its initial `None` value.